batch/batch/worker/worker.py

```diff
@@ -3046,12 +3046,13 @@ async def _initialize_jvms(self):
         while True:
             try:
                 requested_n_cores = self._waiting_for_jvm_with_n_cores.get_nowait()
-                if not self._jvmpools_by_cores[requested_n_cores].full():
+                jvm_pool = self._jvmpools_by_cores[requested_n_cores]
+                if not jvm_pool.full() and jvm_pool.total_jvms_including_borrowed < jvm_pool.max_jvms:
```

Comment:
Isn't this effectively the same test as `not jvm_pool.full()`?

Reply:
Actually yeah I think you're right, will close this, can always resurrect if needed